{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0cacb610",
   "metadata": {},
   "source": [
    "# Analysis of specific junction\n",
    "\n",
    "We are exemplary considering junction CIRMBUYJFK_f__CWCCKOQCWZ_r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "caa44041",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set working directory to project folder\n",
    "import sys, os\n",
    "sys.path.append(os.path.abspath(os.path.join(os.getcwd(), '..')))\n",
    "\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import plotly.graph_objects as go\n",
    "\n",
    "import mplcursors\n",
    "\n",
    "import altair as alt\n",
    "from itertools import combinations\n",
    "import numpy as np\n",
    "import pypangraph as pp\n",
    "from Bio import Phylo, SeqIO, AlignIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "\n",
    "\n",
    "from pathlib import Path\n",
    "import subprocess\n",
    "\n",
    "from Bio.Phylo.TreeConstruction import DistanceCalculator\n",
    "from scipy.cluster.hierarchy import linkage, dendrogram, fcluster\n",
    "from scipy.spatial.distance import squareform\n",
    "\n",
    "from junction_analyis.helpers import get_tree_order\n",
    "import junction_analyis.pangraph_utils as pu\n",
    "from junction_analyis.plotting import plot_junction_pangraph_combined, plot_junction_pangraph_interactive, plot_dendrogram\n",
    "from junction_analyis.consensus import find_consensus_paths, make_deduplicated_paths\n",
    "from junction_analyis.block_alignment import create_block_msas, summarize_block_msas, analyze_alignment, cluster_alignment, retrieve_cluster_members"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "806b02af",
   "metadata": {},
   "source": [
    "## Redefine consensus paths based on sequence information\n",
    "Analyze whether all consensus paths are unique or whether some should be merged."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bc00794",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_junction = \"CIRMBUYJFK_f__CWCCKOQCWZ_r\"\n",
    "example_pangraph = pp.Pangraph.from_json(f\"../results/junction_pangraphs/{example_junction}.json\")\n",
    "\n",
    "consensus_paths, path_dict, similarity_matrix, assignment_df = find_consensus_paths(example_pangraph, rare_block_threshold=5, rare_edge_threshold=5, min_n_isolates_per_consensus=5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ce79b29",
   "metadata": {},
   "outputs": [],
   "source": [
    "block_id = consensus_paths[0].nodes[18].id\n",
    "block_aln = example_pangraph.blocks[block_id].to_biopython_alignment()\n",
    "AlignIO.write(block_aln, f\"../results/marie_playground/{example_junction}_{block_id}_aln.fa\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cdef80e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "block_id = 6932368721519041699\n",
    "block_aln = example_pangraph.blocks[block_id].to_biopython_alignment()\n",
    "AlignIO.write(block_aln, f\"../results/marie_playground/{example_junction}_{block_id}_aln.fa\", \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e252eeb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "block_2_id = consensus_paths[0].nodes[23].id\n",
    "block_ids = [block_id, block_2_id, consensus_paths[0].nodes[6].id, consensus_paths[0].nodes[7].id, consensus_paths[0].nodes[26].id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a3d0f20",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df = pd.read_csv(f\"../results/block_alignments/{example_junction}/{example_junction}_alignment_stats.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae7323b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[summary_df['block_id'].isin(block_ids)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6dede54a",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[summary_df['block_id'] == consensus_paths[0].nodes[16].id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "592d954c",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df.sort_values('mismatch_fraction')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "665bd25b",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df.loc[39, 'block_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "140846da",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_pangraph.to_blockstats_df().loc[summary_df.loc[35, 'block_id']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcabf007",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_pangraph.to_blockcount_df().loc[summary_df.loc[14, 'block_id']][example_pangraph.to_blockcount_df().loc[summary_df.loc[14, 'block_id']] == 1]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b344bdc8",
   "metadata": {},
   "source": [
    "Findings for junction CIRMBUYJFK_f__CWCCKOQCWZ_r:\n",
    "- Even though there are some mismatch columns in the alignment, those are usually point mutations found in one or maximally 2 sequences. These mutations could also stem from sequencing errors.\n",
    "- Block 964531385716605116 (which is found in consensus path 1 only) has a lot of variation, however this stems from only one of the 13 isolates containing this blog\n",
    "- for block 6932368721519041699 there are actually two different versions, it is a duplicated block, it is inserted at many different positions in the genome\n",
    "- Core block 16494405553751752639 seems to have two versions, one version belongs to the short consensus path consensus_3, the other version belongs to the longer consensus path (1 and 2), the other core block 11400043001338627984 has the same two versions as well as all shared blocks\n",
    "    - clustering isolates by core block similarity (e.g. first core block) shows perfect seperation between consensus_3 and consensus_1/2\n",
    "    - wrongly forcing 3 clusters intermixes isolates between consensus_1 and consensus_2\n",
    "    - we should have 2 consensus paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bab90891",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: find a better way to do this (cluster sequences, talk with Marco what would be best)\n",
    "aln_path = f\"../results/block_alignments/{example_junction}/block_11400043001338627984_aln.fa\"\n",
    "distance_matrix, Z, names = cluster_alignment(aln_path)\n",
    "\n",
    "plot_dendrogram(Z, names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e1bc705",
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters = retrieve_cluster_members(Z, names, n_clusters=3)\n",
    "plot_junction_pangraph_interactive(\n",
    "    example_pangraph,\n",
    "    show_consensus=True,\n",
    "    consensus_paths=consensus_paths,\n",
    "    assignments=assignment_df,\n",
    "    order=\"tree\",\n",
    "    highlight_clusters=clusters\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfef58df",
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters_2 = retrieve_cluster_members(Z, names, n_clusters=2)\n",
    "plot_junction_pangraph_interactive(\n",
    "    example_pangraph,\n",
    "    show_consensus=True,\n",
    "    consensus_paths=consensus_paths,\n",
    "    assignments=assignment_df,\n",
    "    order=\"tree\",\n",
    "    highlight_clusters=clusters_2\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ace5536",
   "metadata": {},
   "source": [
    "## Analyze deviations (insertions / deletions) from consensus path\n",
    "Consider the two consensus paths seperately from each other."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7309641",
   "metadata": {},
   "outputs": [],
   "source": [
    "# choosing higher thresholds merges the two consensus paths into one, however this might not be possible for all junctions while still keeping all desired consensus paths\n",
    "\n",
    "consensus_paths, path_dict, similarity_matrix, assignment_df = find_consensus_paths(example_pangraph, rare_block_threshold=10, rare_edge_threshold=10, min_n_isolates_per_consensus=5)\n",
    "plot_junction_pangraph_interactive(\n",
    "    example_pangraph,\n",
    "    show_consensus=True,\n",
    "    consensus_paths=consensus_paths,\n",
    "    assignments=assignment_df,\n",
    "    order=\"tree\",\n",
    "    highlight_clusters=clusters_2\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6dc0a90f",
   "metadata": {},
   "source": [
    "### Consensus 2\n",
    "Consensus path 2 only has two insertion sequences which I manually extracted."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98d0aa79",
   "metadata": {},
   "outputs": [],
   "source": [
    "# as a starting point, start with one insertion sequence\n",
    "\n",
    "# big insertion sequence in NZ_CP098219_1: 2278011600046483881\n",
    "block_id = 2278011600046483881\n",
    "isolate_name = \"NZ_CP098219_1\"\n",
    "\n",
    "def write_block_fasta(example_pangraph, isolate_name, block_id, single_sequence = True):\n",
    "    if single_sequence:\n",
    "        sequence = Seq(example_pangraph.blocks[block_id].to_biopython_records()[0].seq)\n",
    "    else:\n",
    "        sequence = Seq(example_pangraph.blocks[block_id].consensus())\n",
    "    record = SeqRecord(\n",
    "        Seq(example_pangraph.blocks[block_id].to_biopython_records()[0].seq),\n",
    "        id=f\"{isolate_name}|block_{block_id}\",\n",
    "        description=f\"block {block_id} from isolate {isolate_name}\"\n",
    "    )\n",
    "    output_path = f\"../results/atb_lookup/{isolate_name}_block_{block_id}.fasta\"\n",
    "    SeqIO.write(record, output_path, \"fasta\")\n",
    "\n",
    "write_block_fasta(example_pangraph, isolate_name, block_id, single_sequence=True)\n",
    "\n",
    "# results/marie_playground/NZ_CP098219_1_block_2278011600046483881.fasta look up in all the bacteria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1cd3c36f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read results from lexicmap run\n",
    "atb_hits_df = pd.read_csv(\"../results/marie_playground/NZ_CP098219_1_block_2278011600046483881.lexicmap.tsv\", sep=\"\\t\")\n",
    "atb_hits_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a90718e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_sgenome_ids(atb_hits_df, output_file):\n",
    "    sgenome_ids = atb_hits_df.sgenome.to_list()\n",
    "    with open(output_file, \"w\") as f:\n",
    "        for sid in sgenome_ids:\n",
    "            f.write(str(sid) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79cb8df3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# choose best matches and write biosample accession numbers into txt file to look them up in ncbi\n",
    "\n",
    "atb_hits_df = atb_hits_df[atb_hits_df['qcovGnm'] > 99]\n",
    "write_sgenome_ids(atb_hits_df, \"../results/marie_playground/NZ_CP098219_1_block_2278011600046483881.ids.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "579ebe80",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read results from ncbi\n",
    "\n",
    "ncbi_results_df = pd.read_csv(\"../results/marie_playground/NZ_CP098219_1_block_2278011600046483881.ncbi_results.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "307b45c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "atb_hits_info_df = pd.merge(atb_hits_df, ncbi_results_df, on=\"sgenome\", how=\"left\")\n",
    "atb_hits_info_df.to_csv(\"../results/marie_playground/NZ_CP098219_1_block_2278011600046483881.hits_info.tsv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dec45954",
   "metadata": {},
   "outputs": [],
   "source": [
    "atb_hits_info_df.organism.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65e0492c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# redo with another insertion sequence\n",
    "write_block_fasta(example_pangraph, isolate_name = \"NZ_AP022171.1\", block_id = 18228278273347143766, single_sequence=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9be02d7c",
   "metadata": {},
   "source": [
    "sbatch lexicmap_query.sh NZ_CP098219_1_block_2278011600046483881.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d838016",
   "metadata": {},
   "outputs": [],
   "source": [
    "atb_hits_df = pd.read_csv(\"../results/atb_lookup/NZ_AP022171.1_block_18228278273347143766.lexicmap.tsv\", sep=\"\\t\")\n",
    "write_sgenome_ids(atb_hits_df, \"../results/atb_lookup/NZ_AP022171.1_block_18228278273347143766.ids.txt\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62e9f79a",
   "metadata": {},
   "source": [
    "bash fetch_biosample.sh NZ_AP022171.1_block_18228278273347143766.ids.txt NZ_AP022171.1_block_18228278273347143766.ncbi_results.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2727788f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_results_df = pd.read_csv(\"../results/atb_lookup/NZ_AP022171.1_block_18228278273347143766.ncbi_results.tsv\", sep=\"\\t\")\n",
    "atb_hits_info_df = pd.merge(atb_hits_df, ncbi_results_df, on=\"sgenome\", how=\"left\")\n",
    "atb_hits_info_df.to_csv(\"../results/atb_lookup/NZ_AP022171.1_block_18228278273347143766.hits_info.tsv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df194302",
   "metadata": {},
   "outputs": [],
   "source": [
    "atb_hits_info_df.organism.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5558a00e",
   "metadata": {},
   "source": [
    "Insertion sequence:\n",
    "- toxin YdaT family protein, these are often small proteins associated with toxinâ€“antitoxin (TA) systems, used by bacteria to regulate growth or defend against stress and phage infection.\n",
    "- RISPR-Cas-Mediated Gene Silencing Reveals RacR To Be a Negative Regulator of YdaS and YdaT Toxins in Escherichia coli K-12\n",
    "\n",
    "### Consensus 1\n",
    "Derive all blocks with corresponding isolate name that deviate from the consensus path."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "068b1660",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get a list of all blocks that deviate from consensus paths (consider insertions and deletions!)\n",
    "\n",
    "# get isolates belonging to consensus 1\n",
    "isolates_1 = assignment_df[assignment_df['best_consensus'] == 'consensus_1'].index.tolist()\n",
    "isolates_1_set = set(isolates_1)\n",
    "\n",
    "# make path dict\n",
    "path_dict = example_pangraph.to_path_dictionary()\n",
    "path_dict = {isolate: pu.Path.from_tuple_list(path, 'node') for isolate, path in path_dict.items() if isolate in isolates_1_set}\n",
    "\n",
    "# deduplicate blocks\n",
    "blockstats_df = example_pangraph.to_blockstats_df()\n",
    "deduplicated_paths, deduplicated_blog_freq = make_deduplicated_paths(blockstats_df, path_dict)\n",
    "\n",
    "# compare deduplicated paths to consensus paths to find deviations, consensus paths are already deduplicated\n",
    "\n",
    "\n",
    "# insertions\n",
    "# start with easier one\n",
    "consensus_paths[0]\n",
    "\n",
    "\n",
    "\n",
    "# potentially convert isolates to path data strucutes\n",
    "example_pangraph.paths[isolates_1[0]]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9549dea5",
   "metadata": {},
   "outputs": [],
   "source": [
    "consensus_paths[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca239a38",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_dict"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "junctions",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
