{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0cacb610",
   "metadata": {},
   "source": [
    "# Analysis of specific junction\n",
    "\n",
    "We are exemplary considering junction CIRMBUYJFK_f__CWCCKOQCWZ_r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "caa44041",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set working directory to project folder\n",
    "import sys, os, re\n",
    "sys.path.append(os.path.abspath(os.path.join(os.getcwd(), '..')))\n",
    "\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import plotly.graph_objects as go\n",
    "\n",
    "import mplcursors\n",
    "\n",
    "import altair as alt\n",
    "from itertools import combinations\n",
    "import numpy as np\n",
    "import pypangraph as pp\n",
    "from Bio import Phylo, SeqIO, AlignIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "\n",
    "import plotly.io as pio\n",
    "pio.renderers.default = \"browser\" \n",
    "\n",
    "\n",
    "from pathlib import Path\n",
    "import subprocess\n",
    "\n",
    "from Bio.Phylo.TreeConstruction import DistanceCalculator\n",
    "from scipy.cluster.hierarchy import linkage, dendrogram, fcluster\n",
    "from scipy.spatial.distance import squareform\n",
    "\n",
    "from junction_analysis.helpers import get_tree_order, convert_gbk_fasta\n",
    "import junction_analysis.pangraph_utils as pu\n",
    "from junction_analysis.plotting import plot_junction_pangraph_combined, plot_junction_pangraph_interactive, plot_dendrogram\n",
    "from junction_analysis.consensus import find_consensus_paths, make_deduplicated_paths\n",
    "from junction_analysis.block_alignment import create_block_msas, summarize_block_msas, analyze_alignment, cluster_alignment, retrieve_cluster_assignments\n",
    "\n",
    "from junction_analysis.annotate_insertions import write_insertions_fasta, get_insertions_deletions_from_consensus, retrieve_SAMids_txt, combine_NCBI_atb_results, find_insertion_hits_own_genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bab90891",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_junction = \"CIRMBUYJFK_f__CWCCKOQCWZ_r\"\n",
    "example_pangraph = pp.Pangraph.from_json(f\"../results/junction_pangraphs/{example_junction}.json\")\n",
    "\n",
    "aln_path = f\"../results/block_alignments/{example_junction}/block_11400043001338627984_aln.fa\"\n",
    "distance_matrix, Z, names = cluster_alignment(aln_path)\n",
    "clusters = retrieve_cluster_assignments(Z, names, n_clusters=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ace5536",
   "metadata": {},
   "source": [
    "## Analyze deviations (insertions / deletions) from consensus path\n",
    "Consider the two consensus paths seperately from each other."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a7309641",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2 unique paths.\n"
     ]
    }
   ],
   "source": [
    "# choosing higher thresholds merges the two consensus paths into one, however this might not be possible for all junctions while still keeping all desired consensus paths\n",
    "consensus_paths, path_dict, similarity_matrix, assignment_df = find_consensus_paths(example_pangraph, rare_block_threshold=10, rare_edge_threshold=10, min_n_isolates_per_consensus=5)\n",
    "plot_junction_pangraph_interactive(\n",
    "    example_pangraph,\n",
    "    show_consensus=True,\n",
    "    consensus_paths=consensus_paths,\n",
    "    assignments=assignment_df,\n",
    "    order=\"tree\",\n",
    "    cluster_map=clusters,\n",
    "    title = \"Junction Block Structure with 2 Clusters\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22eb9513",
   "metadata": {},
   "source": [
    "### Consensus 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ae3489f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Insertions:\n",
      "NZ_AP022171.1 INSERTED: [18228278273347143766|+|]\n",
      "NZ_CP098219.1 INSERTED: [2278011600046483881|+|]\n",
      "\n",
      "Deletions:\n"
     ]
    }
   ],
   "source": [
    "insertions_2, deletions_2 = get_insertions_deletions_from_consensus(example_pangraph, assignment_df, consensus_paths, consensus = 2)\n",
    "write_insertions_fasta(example_junction, insertions_2, consensus = 2)\n",
    "\n",
    "# upload fastas: scp -r /Users/mariebecker/Documents/Uni/ETH/Masterarbeit/marie-junctions/results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2 becker0010@transfer12.scicore.unibas.ch:/scicore/home/neher/GROUP/data/2025_all_the_bacteria/lexicmap_index/queries/CIRMBUYJFK_f__CWCCKOQCWZ_r/\n",
    "# run:  bash run_all_lexicmap.sh ./queries/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2 on cluster\n",
    "# download results:  scp -r becker0010@transfer12.scicore.unibas.ch:'/scicore/home/neher/GROUP/data/2025_all_the_bacteria/lexicmap_index/queries/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2/*.lexicmap.tsv' /Users/mariebecker/Documents/Uni/ETH/Masterarbeit/marie-junctions/results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dad9c3ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "# retrieve SAM... ids for NCBI lookup\n",
    "retrieve_SAMids_txt(parent_dir=\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2\")\n",
    "\n",
    "# to lookup in NCBI run from results/atb_lookup: bash fetch_biosamples_consensus.sh CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "ad0c67ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "combine_NCBI_atb_results(parent_dir=\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41aecac5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "organism\n",
       "Escherichia coli                  187\n",
       "Escherichia coli O25b:H4-ST131      4\n",
       "Acinetobacter baumannii             1\n",
       "Campylobacter jejuni                1\n",
       "Escherichia coli KTE216             1\n",
       "Escherichia coli O13/O135:H4        1\n",
       "Shigella sonnei                     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# not annotated as insertion sequence but this is gene content annotation:\n",
    "#NZ_AP022171.1\tGenBank\tgene\t10173\t10694\t.\t+\t.\tID=H7R27_RS10275;Name=ugpC;gene=ugpC;is_partial=False;locus_tag=H7R27_RS10275;pseudo=\n",
    "#NZ_AP022171.1\tGenBank\tCDS\t10173\t10694\t.\t+\t.\tID=H7R27_RS10275;Name=ugpC;codon_start=1;gene=ugpC;inference=COORDINATES: similar to AA sequence:RefSeq:NP_312323.2;is_partial=False;locus_tag=H7R27_RS10275;note=incomplete%3B partial in the middle of a contig%3B missing N-terminus%3B Derived by automated computational analysis using gene prediction method: Protein Homology.;product=sn-glycerol 3-phosphate ABC transporter ATP binding protein UgpC;pseudo=;transl_table=11\n",
    "#NZ_AP022171.1\tGenBank\tgene\t10691\t11008\t.\t+\t.\tID=H7R27_RS10280;Name=H7R27_RS10280;is_partial=False;locus_tag=H7R27_RS10280;old_locus_tag=WP5S18E09_20050;pseudo=\n",
    "#NZ_AP022171.1\tGenBank\tCDS\t10691\t11008\t.\t+\t.\tID=H7R27_RS10280;Name=H7R27_RS10280;go_function=GO:0008081 - phosphoric diester hydrolase activity [Evidence IEA];go_process=GO:0006629 - lipid metabolic process [Evidence IEA];codon_start=1;inference=COORDINATES: similar to AA sequence:RefSeq:NP_709219.1;is_partial=False;locus_tag=H7R27_RS10280;note=incomplete%3B partial in the middle of a contig%3B missing C-terminus%3B Derived by automated computational analysis using gene prediction method: Protein Homology.;old_locus_tag=WP5S18E09_20050;product=glycerophosphodiester phosphodiesterase family protein;pseudo=;transl_table=11\n",
    "atb_hits_info_df = pd.read_csv(\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2/NZ_AP022171.1_segment_0.hits_info.tsv\", sep = \"\\t\")\n",
    "atb_hits_info_df.value_counts(\"organism\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78c9087c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "organism\n",
       "Escherichia coli                       83\n",
       "Escherichia coli O145:H28               3\n",
       "Salmonella enterica                     2\n",
       "Salmonella enterica subsp. enterica     1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# long insertion: prophage\n",
    "atb_hits_info_df = pd.read_csv(\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2/NZ_CP098219.1_segment_0.hits_info.tsv\", sep = \"\\t\")\n",
    "atb_hits_info_df.value_counts(\"organism\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7b5d5305",
   "metadata": {},
   "source": [
    "### Consensus 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "068b1660",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Insertions:\n",
      "NZ_OX030701.1 INSERTED: [6932368721519041699|-|1314867519434933921]\n",
      "NZ_CP021935.1 INSERTED: [517157241956952768|+|]\n",
      "NZ_CP076693.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_CP128947.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_CP107114.1 INSERTED: [517157241956952768|+|]\n",
      "NZ_CP107172.1 INSERTED: [517157241956952768|+|]\n",
      "NZ_CP030337.1 INSERTED: [6932368721519041699|+|16056383497489254679]\n",
      "NZ_CP124429.1 INSERTED: [517157241956952768|-|]\n",
      "NZ_CP018970.1 INSERTED: [6932368721519041699|-|17115477248436968411]\n",
      "NZ_CP019015.1 INSERTED: [16982887268023976258|-|]_[964531385716605116|-|]\n",
      "NZ_CP124460.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_AP022044.1 INSERTED: [6932368721519041699|+|12883103544713419939]\n",
      "NZ_AP022044.1 INSERTED: [6932368721519041699|-|14506298656834147691]\n",
      "NZ_CP103562.1 INSERTED: [7844090243072536214|+|]\n",
      "NZ_CP051661.1 INSERTED: [6932368721519041699|+|12883103544713419939]\n",
      "NZ_CP051661.1 INSERTED: [6932368721519041699|+|10145330714100251892]\n",
      "NZ_CP124455.1 INSERTED: [6932368721519041699|-|15558722379768673797]\n",
      "NZ_CP023826.1 INSERTED: [6932368721519041699|-|17115477248436968411]\n",
      "NZ_CP019012.1 INSERTED: [6932368721519041699|-|10145330714100251892]\n",
      "NZ_CP036245.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_CP059130.1 INSERTED: [517157241956952768|+|]\n",
      "NZ_CP059130.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_CP027534.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_CP018979.1 INSERTED: [517157241956952768|-|]\n",
      "NZ_CP134384.1 INSERTED: [185719230501067033|+|]\n",
      "NZ_CP019008.1 INSERTED: [185719230501067033|+|]\n",
      "NZ_CP124410.1 INSERTED: [6932368721519041699|+|9834685609301298901]\n",
      "NZ_CP107122.1 INSERTED: [6932368721519041699|-|15558722379768673797]\n",
      "NZ_CP107122.1 INSERTED: [6932368721519041699|+|11426489427845291638]\n",
      "NZ_CP051663.1 INSERTED: [6932368721519041699|+|7987278316461198430]\n",
      "NZ_CP107162.1 INSERTED: [6932368721519041699|-|15558722379768673797]\n",
      "NZ_CP124374.1 INSERTED: [16982887268023976258|-|]_[964531385716605116|-|]\n",
      "NZ_CP124424.1 INSERTED: [964531385716605116|+|]\n",
      "NZ_AP026794.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_AP026794.1 INSERTED: [16556873922195442803|+|]\n",
      "NZ_CP021454.1 INSERTED: [6932368721519041699|-|913783977968963568]\n",
      "NZ_AP026788.1 INSERTED: [964531385716605116|+|]_[16982887268023976258|+|]\n",
      "NZ_AP026788.1 INSERTED: [16556873922195442803|+|]\n",
      "NZ_CP124467.1 INSERTED: [185719230501067033|+|]\n",
      "NZ_AP022326.1 INSERTED: [6932368721519041699|+|5100696243500885373]\n",
      "NZ_AP022326.1 INSERTED: [6932368721519041699|+|6097212296783908714]\n",
      "NZ_CP021179.1 INSERTED: [517157241956952768|+|]\n",
      "NZ_CP010876.1 INSERTED: [16982887268023976258|-|]_[964531385716605116|-|]\n",
      "NZ_CP133923.1 INSERTED: [185719230501067033|+|]\n",
      "NZ_CP051659.1 INSERTED: [6932368721519041699|+|10145330714100251892]\n",
      "NZ_LR890693.1 INSERTED: [6932368721519041699|-|11588582069103229002]\n",
      "NZ_LR890693.1 INSERTED: [16982887268023976258|-|]_[964531385716605116|-|]\n",
      "NZ_CP128907.1 INSERTED: [6932368721519041699|+|1708810210093552769]\n",
      "NZ_CP107151.1 INSERTED: [6932368721519041699|-|15558722379768673797]\n",
      "\n",
      "Deletions:\n",
      "NZ_CP018970.1 DELETED: [13879318911159590194|+|]_[11455254306155687211|+|]\n",
      "NZ_AP022044.1 DELETED: [11426489427845291638|+|]_[913783977968963568|+|]_[9242629882957449119|+|]\n",
      "NZ_CP019012.1 DELETED: [12055620567302113403|+|]\n",
      "NZ_CP134384.1 DELETED: [12883103544713419939|+|]_[11588582069103229002|+|]_[7754538651979209778|+|]\n",
      "NZ_CP019008.1 DELETED: [12883103544713419939|+|]_[11588582069103229002|+|]_[7754538651979209778|+|]\n",
      "NZ_CP051663.1 DELETED: [5100696243500885373|+|]_[15395175763379073292|+|]_[14506298656834147691|+|]_[11426489427845291638|+|]_[913783977968963568|+|]_[9242629882957449119|+|]_[1314867519434933921|+|]_[3035246746052306249|+|]_[16543294998939926181|+|]_[1708810210093552769|+|]_[17115477248436968411|+|]_[13879318911159590194|+|]_[11455254306155687211|+|]_[6304055291150384278|+|]_[6097212296783908714|+|]_[10145330714100251892|+|]\n",
      "NZ_AP026788.1 DELETED: [6097212296783908714|+|]\n",
      "NZ_CP124467.1 DELETED: [12883103544713419939|+|]_[11588582069103229002|+|]_[7754538651979209778|+|]\n",
      "NZ_CP133923.1 DELETED: [12883103544713419939|+|]_[11588582069103229002|+|]_[7754538651979209778|+|]\n",
      "NZ_CP128907.1 DELETED: [17115477248436968411|+|]_[13879318911159590194|+|]_[11455254306155687211|+|]_[6304055291150384278|+|]_[6097212296783908714|+|]\n"
     ]
    }
   ],
   "source": [
    "# get a list of all blocks that deviate from consensus paths (consider insertions and deletions!)\n",
    "\n",
    "insertions, deletions = get_insertions_deletions_from_consensus(example_pangraph, assignment_df, consensus_paths, consensus = 1)\n",
    "write_insertions_fasta(example_junction, insertions, consensus = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80df2ed7",
   "metadata": {},
   "source": [
    "run lexicmap search on cluster for all sequences: bash run_all_lexicmap.sh ./queries/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7370d91a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# retrieve SAM... to look up in NCBI:\n",
    "retrieve_SAMids_txt(parent_dir=\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11b7284b",
   "metadata": {},
   "source": [
    "retrieve info from NCBI: bash fetch_biosamples_consensus1.sh out of this folder results/atb_lookup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7a4e446",
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge results\n",
    "combine_NCBI_atb_results(parent_dir=\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "57b4a5c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "organism\n",
       "Campylobacter jejuni         9\n",
       "Escherichia coli             2\n",
       "Helicobacter winghamensis    1\n",
       "Klebsiella pneumoniae        1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP030337.1_segment_0.ncbi_results.tsv\", sep=\"\\t\").value_counts('organism')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5416213",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO: should I run blocks seperately or insertions as a whole\n",
    "# TODO: rerun lexicmap ... now with correctly inverted sequences"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61d8d7cd",
   "metadata": {},
   "source": [
    "Information about different insertion sequences:\n",
    "\n",
    "- 5 different versions of this block, if a block is twice in one sequence it is always the same version twice, versions can be inserted or not inserted, found between 1 and 9 times in other genomes\n",
    "    - NZ_CP107151.1, blocks [6932368721519041699|-|]: majority of hits are Mycoplasma mycoides or E. coli and some others, IS1 based on annotation, same insertion in NZ_CP107162.1 and NZ_CP107122.1 blocks [6932368721519041699|-|], same hits, NZ_CP107122.1 even has same insertion a second time[6932368721519041699|+|], NZ_CP128907.1|segment_0 path[6932368721519041699|+|1708810210093552769] length760, NZ_AP022326.1|segment_0 path[6932368721519041699|+|5100696243500885373] length777, NZ_AP022326.1|segment_1 path[6932368721519041699|+|6097212296783908714] length777, NZ_CP124410.1: path[6932368721519041699|+|] IS1, Mycoplasma mycoides and E.coli and Campylobacter jejuni and some more\n",
    "    - NZ_CP021454.1|segment_0 path[6932368721519041699|-|913783977968963568] length777 and NZ_CP019012.1|segment_0 path[6932368721519041699|-|10145330714100251892] length768 (only found once in genome), inverted, IS1, only 4 hits and different ones --> difference within block!\n",
    "    - NZ_OX030701.1|segment_0 path[6932368721519041699|-|] length760, NZ_CP018970.1|segment_0 path[6932368721519041699|-|17115477248436968411] length767, NZ_CP023826.1|segment_0 path[6932368721519041699|-|17115477248436968411] length767, NZ_LR890693.1|segment_0 path[6932368721519041699|-|11588582069103229002] length760, NZ_CP051661.1|segment_1 path[6932368721519041699|+|10145330714100251892] length776 and segment_0 path[6932368721519041699|+|12883103544713419939] length776, NZ_CP051663.1|segment_0 path[6932368721519041699|+|7987278316461198430] length764, NZ_CP051659.1|segment_0 path[6932368721519041699|+|10145330714100251892] length760, NZ_CP018970.1|segment_0 path[6932368721519041699|-|17115477248436968411] length767, NZ_CP023826.1|segment_0 path[6932368721519041699|-|17115477248436968411] length767, NZ_CP124455.1|segment_0 path[6932368721519041699|-|15558722379768673797] length768, same hits, less hits, IS1\n",
    "    - NZ_CP030337.1|segment_0 path[6932368721519041699|+|16056383497489254679] length760, Campylobacter jejuni version (?)\n",
    "    - NZ_AP022044.1|segment_0 path[6932368721519041699|+|12883103544713419939] length767 and NZ_AP022044.1|segment_1 path[6932368721519041699|-|14506298656834147691] length767, new version, mostly E.coli, 25 hits\n",
    "\n",
    "- NZ_CP103562.1: path[7844090243072536214|+|] E. coli and some single different ones, IS3, found twice in genome\n",
    "\n",
    "- double block: all found twice (once elsewhere in genome), all at same position, just some inverted\n",
    "    - NZ_CP124374.1 and NZ_CP019015.1: [16982887268023976258|-|]_[964531385716605116|-|] ISL3 (both blogs together): no hits (TODO: change parameters and rerun or run blogs seperately)\n",
    "    - NZ_CP059130.1 (segment 1), NZ_AP026788.1 and NZ_AP026794.1, [964531385716605116|+|]_[16982887268023976258|+|], NZ_CP124460.1|segment_0 path[964531385716605116|+|]_[16982887268023976258|+|] length1863, NZ_CP027534.1|segment_0 path[964531385716605116|+|]_[16982887268023976258|+|] length1863, NZ_CP128947.1|segment_0 path[964531385716605116|+|]_[16982887268023976258|+|] length1863: ISL3, mostly E.coli, same hits\n",
    "    - NZ_LR890693.1|segment_1 path[964531385716605116|-|]_[16982887268023976258|-|], NZ_CP010876.1|segment_0 path[964531385716605116|-|]_[16982887268023976258|-|] length1893, (order was inverted here) length1893, this one got hits now, all E.coli (why hits even though sequences don't fit)\n",
    "- NZ_CP124424.1|segment_0 path[964531385716605116|+|] length1324, exist as partly insertion (only first block), less hits, E.coli and Lelliottia amnigena, ISL3, found twice\n",
    "\n",
    "\n",
    "- NZ_CP021935.1, NZ_CP059130.1 (segment 0), NZ_CP107114.1 and NZ_CP107172.1 (identical), NZ_CP107114.1, NZ_CP107172, NZ_CP021179.1, NZ_CP018979.1 and NZ_CP124429.1 (inverted): 517157241956952768 IS3, 100% hits in E.coli and in other bacteria, mostly E. coli, all get the same hits, they might differ in a few bases or some bases at ends, inserted at different positions in genome, one has 3 base overhang, apart from that they are completely the same, different amount of hits in genome (between 2 and 7)\n",
    "\n",
    "- NZ_AP026788.1 and NZ_AP026794.1: path[16556873922195442803|+|] IS1380, E. coli, Klebsiella pneumoniae, Pseudomonas aeruginosa,Salmonella enterica all frequent, many more, twice in genome and both have same hits\n",
    "\n",
    "- NZ_CP019008.1|segment_0 path[185719230501067033|+|] length4204, NZ_CP133923.1|segment_0 path[185719230501067033|+|] length4204, NZ_CP124467.1|segment_0 path[185719230501067033|+|] length4204, NZ_CP134384.1|segment_0 path[185719230501067033|+|] length4203 (one nucleotide shorter): 200 hits that are all E. coli. all sequences identical, only once in genome\n",
    "\n",
    "\n",
    "Difference between consensus 1 and consensus 2 in one big prophage according to annotations\n",
    "\n",
    "Be careful to not confuse context for next block!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "25d70054",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>qlen</th>\n",
       "      <th>hits</th>\n",
       "      <th>sgenome</th>\n",
       "      <th>qcovGnm</th>\n",
       "      <th>cls</th>\n",
       "      <th>hsp</th>\n",
       "      <th>qcovHSP</th>\n",
       "      <th>alenHSP</th>\n",
       "      <th>pident</th>\n",
       "      <th>gaps</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "      <th>organism</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>sequence_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA4062129</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN14049574</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>E. coli QEC12-15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN07508953</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>29-KPC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA104040230</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111271919</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN08382692</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>GASRECE110</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111273518</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN18977267</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>missing</td>\n",
       "      <td>missing</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111273646</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111273548</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111273647</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111273667</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN01922332</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli ADL-327</td>\n",
       "      <td>ADL-327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN15904810</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>hkcpe141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN26379194</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>PFEEC0739</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111274498</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA111274494</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMD00500056</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>JBCDAFJ-19-0051</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMD00497908</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>JBAJAAH-19-0014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN15904789</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Klebsiella quasipneumoniae</td>\n",
       "      <td>hkcpe120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA6155701</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.870</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1389</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMEA7230815</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>99.740</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1386</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN30575479</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>98.698</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1350</td>\n",
       "      <td>Enterobacter cloacae</td>\n",
       "      <td>missing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN26796500</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>98.307</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1335</td>\n",
       "      <td>Enterobacter cloacae</td>\n",
       "      <td>Enterobacter_cloacae_P36220121</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NZ_AP022044.1|segment_0</td>\n",
       "      <td>767</td>\n",
       "      <td>25</td>\n",
       "      <td>SAMN41034631</td>\n",
       "      <td>100.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>100.0</td>\n",
       "      <td>768</td>\n",
       "      <td>98.177</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1332</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>VUTI177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      query  qlen  hits         sgenome  qcovGnm  cls  hsp  \\\n",
       "0   NZ_AP022044.1|segment_0   767    25    SAMEA4062129    100.0    1    1   \n",
       "1   NZ_AP022044.1|segment_0   767    25    SAMN14049574    100.0    1    1   \n",
       "2   NZ_AP022044.1|segment_0   767    25    SAMN07508953    100.0    1    1   \n",
       "3   NZ_AP022044.1|segment_0   767    25  SAMEA104040230    100.0    1    1   \n",
       "4   NZ_AP022044.1|segment_0   767    25  SAMEA111271919    100.0    1    1   \n",
       "5   NZ_AP022044.1|segment_0   767    25    SAMN08382692    100.0    1    1   \n",
       "6   NZ_AP022044.1|segment_0   767    25  SAMEA111273518    100.0    1    1   \n",
       "7   NZ_AP022044.1|segment_0   767    25    SAMN18977267    100.0    1    1   \n",
       "8   NZ_AP022044.1|segment_0   767    25  SAMEA111273646    100.0    1    1   \n",
       "9   NZ_AP022044.1|segment_0   767    25  SAMEA111273548    100.0    1    1   \n",
       "10  NZ_AP022044.1|segment_0   767    25  SAMEA111273647    100.0    1    1   \n",
       "11  NZ_AP022044.1|segment_0   767    25  SAMEA111273667    100.0    1    1   \n",
       "12  NZ_AP022044.1|segment_0   767    25    SAMN01922332    100.0    1    1   \n",
       "13  NZ_AP022044.1|segment_0   767    25    SAMN15904810    100.0    1    1   \n",
       "14  NZ_AP022044.1|segment_0   767    25    SAMN26379194    100.0    1    1   \n",
       "15  NZ_AP022044.1|segment_0   767    25  SAMEA111274498    100.0    1    1   \n",
       "16  NZ_AP022044.1|segment_0   767    25  SAMEA111274494    100.0    1    1   \n",
       "17  NZ_AP022044.1|segment_0   767    25    SAMD00500056    100.0    1    1   \n",
       "18  NZ_AP022044.1|segment_0   767    25    SAMD00497908    100.0    1    1   \n",
       "19  NZ_AP022044.1|segment_0   767    25    SAMN15904789    100.0    1    1   \n",
       "20  NZ_AP022044.1|segment_0   767    25    SAMEA6155701    100.0    1    1   \n",
       "21  NZ_AP022044.1|segment_0   767    25    SAMEA7230815    100.0    1    1   \n",
       "22  NZ_AP022044.1|segment_0   767    25    SAMN30575479    100.0    1    1   \n",
       "23  NZ_AP022044.1|segment_0   767    25    SAMN26796500    100.0    1    1   \n",
       "24  NZ_AP022044.1|segment_0   767    25    SAMN41034631    100.0    1    1   \n",
       "\n",
       "    qcovHSP  alenHSP  pident  gaps  evalue  bitscore  \\\n",
       "0     100.0      768  99.870     1     0.0      1389   \n",
       "1     100.0      768  99.870     1     0.0      1389   \n",
       "2     100.0      768  99.870     1     0.0      1389   \n",
       "3     100.0      768  99.870     1     0.0      1389   \n",
       "4     100.0      768  99.870     1     0.0      1389   \n",
       "5     100.0      768  99.870     1     0.0      1389   \n",
       "6     100.0      768  99.870     1     0.0      1389   \n",
       "7     100.0      768  99.870     1     0.0      1389   \n",
       "8     100.0      768  99.870     1     0.0      1389   \n",
       "9     100.0      768  99.870     1     0.0      1389   \n",
       "10    100.0      768  99.870     1     0.0      1389   \n",
       "11    100.0      768  99.870     1     0.0      1389   \n",
       "12    100.0      768  99.870     1     0.0      1389   \n",
       "13    100.0      768  99.870     1     0.0      1389   \n",
       "14    100.0      768  99.870     1     0.0      1389   \n",
       "15    100.0      768  99.870     1     0.0      1389   \n",
       "16    100.0      768  99.870     1     0.0      1389   \n",
       "17    100.0      768  99.870     1     0.0      1389   \n",
       "18    100.0      768  99.870     1     0.0      1389   \n",
       "19    100.0      768  99.870     1     0.0      1389   \n",
       "20    100.0      768  99.870     1     0.0      1389   \n",
       "21    100.0      768  99.740     1     0.0      1386   \n",
       "22    100.0      768  98.698     1     0.0      1350   \n",
       "23    100.0      768  98.307     1     0.0      1335   \n",
       "24    100.0      768  98.177     1     0.0      1332   \n",
       "\n",
       "                      organism                          strain  serovar  \\\n",
       "0             Escherichia coli                             NaN      NaN   \n",
       "1             Escherichia coli                E. coli QEC12-15      NaN   \n",
       "2             Escherichia coli                          29-KPC      NaN   \n",
       "3             Escherichia coli                             NaN      NaN   \n",
       "4             Escherichia coli                             NaN      NaN   \n",
       "5             Escherichia coli                      GASRECE110      NaN   \n",
       "6             Escherichia coli                             NaN      NaN   \n",
       "7             Escherichia coli                         missing  missing   \n",
       "8             Escherichia coli                             NaN      NaN   \n",
       "9             Escherichia coli                             NaN      NaN   \n",
       "10            Escherichia coli                             NaN      NaN   \n",
       "11            Escherichia coli                             NaN      NaN   \n",
       "12    Escherichia coli ADL-327                         ADL-327      NaN   \n",
       "13            Escherichia coli                        hkcpe141      NaN   \n",
       "14            Escherichia coli                       PFEEC0739      NaN   \n",
       "15            Escherichia coli                             NaN      NaN   \n",
       "16            Escherichia coli                             NaN      NaN   \n",
       "17            Escherichia coli                 JBCDAFJ-19-0051      NaN   \n",
       "18            Escherichia coli                 JBAJAAH-19-0014      NaN   \n",
       "19  Klebsiella quasipneumoniae                        hkcpe120      NaN   \n",
       "20            Escherichia coli                             NaN      NaN   \n",
       "21            Escherichia coli                             NaN      NaN   \n",
       "22        Enterobacter cloacae                         missing      NaN   \n",
       "23        Enterobacter cloacae  Enterobacter_cloacae_P36220121      NaN   \n",
       "24            Escherichia coli                         VUTI177      NaN   \n",
       "\n",
       "    sequence_type  \n",
       "0             NaN  \n",
       "1             NaN  \n",
       "2             NaN  \n",
       "3             NaN  \n",
       "4             NaN  \n",
       "5             NaN  \n",
       "6             NaN  \n",
       "7             NaN  \n",
       "8             NaN  \n",
       "9             NaN  \n",
       "10            NaN  \n",
       "11            NaN  \n",
       "12            NaN  \n",
       "13            NaN  \n",
       "14            NaN  \n",
       "15            NaN  \n",
       "16            NaN  \n",
       "17            NaN  \n",
       "18            NaN  \n",
       "19            NaN  \n",
       "20            NaN  \n",
       "21            NaN  \n",
       "22            NaN  \n",
       "23            NaN  \n",
       "24            NaN  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP022044.1_segment_0.hits_info.tsv\", sep=\"\\t\")[[\"query\", \"qlen\", 'hits', 'sgenome', 'qcovGnm', 'cls', 'hsp',\n",
    "       'qcovHSP', 'alenHSP', 'pident', 'gaps', 'evalue', 'bitscore', 'organism', 'strain',\n",
    "       'serovar', 'sequence_type']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8cfd13e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'GGTAATGACTCCAACTTACTGATAGTGTTTTATGTTCAGATAATGCCCGATGACTTTGTCATGCAGCTCCACCGATTTTGAGAACGACAGTGACTTCCGTCCCAGCCTTGCCAGATGTTGTCTCAGATTCAGGTTATGTCGCTCAATGCGCTGAGTGTAACGCTTGCTGATTACGTGCAGCTTTCCCTTCAGGCGGGATTCATACAGCGGCCAGCCATCCGTCATCCATACCACGACCTCAAAGGCCGACAGCAGGCTCAGAAGACGCTCCAGTGTGGCCAGAGTGCGTTCACCGAAGACGTGCGCCACAACCGTCCTCCGTATCCTGTCATACGCGTAAAACAGCCAGCGCTGACGTGATTTAGCACCGACGTAGCCCCACTGTTCGTCCATTTCAGCGCAGACAATCACATCACTGCCCGGCTGTATGCGCGAGGTTACCGACTGCGGCCTGAGTTTTTTAAGTGACGTAAAATCGTGTTGAGGCCAACGCCCATAATGCGGGCGGTTGCCCGGCATCCAACACCATTCATGGCCATATCAATGATTTTCTGGTGTGTACCGGGTTGAGAAGCGGTGTAAGTGAACTGCAGTTGCCATGTTTTACGGCAGTGAGAGCAGAGATAGCGCTGATGTCCGGCAGTACTTTTACCGTTACGCACCACGCCTTCAGTAGCTGAGCAGGAGGGACAACTGATGGAGATGGAAGCCACGGGAGCACCTCAAAAACACCATCATACACTAAATCAGTAAGTTGGCA'"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# for deletions, retrieve block consensus sequence to search in atb database\n",
    "# TODO: maybe find a better way that takes all isolates beloning to consensus and containing the block and recalculate consensus sequence (talk to Marco)\n",
    "# TODO: do I also search deletions in the database?"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18851c30",
   "metadata": {},
   "source": [
    "## Search insertion sequences in their own genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "29692ad5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converting: ../data/gbk/NZ_CP128947.1.gbk → ../results/genome_fastas/NZ_CP128947.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107182.1.gbk → ../results/genome_fastas/NZ_CP107182.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124367.1.gbk → ../results/genome_fastas/NZ_CP124367.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124322.1.gbk → ../results/genome_fastas/NZ_CP124322.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP027534.1.gbk → ../results/genome_fastas/NZ_CP027534.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107142.1.gbk → ../results/genome_fastas/NZ_CP107142.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP010876.1.gbk → ../results/genome_fastas/NZ_CP010876.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP116085.1.gbk → ../results/genome_fastas/NZ_CP116085.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP133927.1.gbk → ../results/genome_fastas/NZ_CP133927.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP022044.1.gbk → ../results/genome_fastas/NZ_AP022044.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP095137.1.gbk → ../results/genome_fastas/NZ_CP095137.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP021454.1.gbk → ../results/genome_fastas/NZ_CP021454.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP069589.1.gbk → ../results/genome_fastas/NZ_CP069589.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124343.1.gbk → ../results/genome_fastas/NZ_CP124343.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP116181.1.gbk → ../results/genome_fastas/NZ_CP116181.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124429.1.gbk → ../results/genome_fastas/NZ_CP124429.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124341.1.gbk → ../results/genome_fastas/NZ_CP124341.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP015085.1.gbk → ../results/genome_fastas/NZ_CP015085.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP063515.1.gbk → ../results/genome_fastas/NZ_CP063515.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP115390.1.gbk → ../results/genome_fastas/NZ_CP115390.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107164.1.gbk → ../results/genome_fastas/NZ_CP107164.1.fasta\n",
      "Converting: ../data/gbk/NZ_OX637961.1.gbk → ../results/genome_fastas/NZ_OX637961.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124339.1.gbk → ../results/genome_fastas/NZ_CP124339.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP082831.1.gbk → ../results/genome_fastas/NZ_CP082831.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP089445.1.gbk → ../results/genome_fastas/NZ_CP089445.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP080118.1.gbk → ../results/genome_fastas/NZ_CP080118.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124475.1.gbk → ../results/genome_fastas/NZ_CP124475.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP022326.1.gbk → ../results/genome_fastas/NZ_AP022326.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124320.1.gbk → ../results/genome_fastas/NZ_CP124320.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP021207.1.gbk → ../results/genome_fastas/NZ_CP021207.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124471.1.gbk → ../results/genome_fastas/NZ_CP124471.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP023853.1.gbk → ../results/genome_fastas/NZ_CP023853.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP072980.1.gbk → ../results/genome_fastas/NZ_CP072980.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP018979.1.gbk → ../results/genome_fastas/NZ_CP018979.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124324.1.gbk → ../results/genome_fastas/NZ_CP124324.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107184.1.gbk → ../results/genome_fastas/NZ_CP107184.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP049077.2.gbk → ../results/genome_fastas/NZ_CP049077.2.fasta\n",
      "Converting: ../data/gbk/NZ_LS992190.1.gbk → ../results/genome_fastas/NZ_LS992190.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103562.1.gbk → ../results/genome_fastas/NZ_CP103562.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124495.1.gbk → ../results/genome_fastas/NZ_CP124495.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124345.1.gbk → ../results/genome_fastas/NZ_CP124345.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124455.1.gbk → ../results/genome_fastas/NZ_CP124455.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124410.1.gbk → ../results/genome_fastas/NZ_CP124410.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP095520.1.gbk → ../results/genome_fastas/NZ_CP095520.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124385.1.gbk → ../results/genome_fastas/NZ_CP124385.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP097367.1.gbk → ../results/genome_fastas/NZ_CP097367.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107162.1.gbk → ../results/genome_fastas/NZ_CP107162.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124387.1.gbk → ../results/genome_fastas/NZ_CP124387.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP116145.1.gbk → ../results/genome_fastas/NZ_CP116145.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP133923.1.gbk → ../results/genome_fastas/NZ_CP133923.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124347.1.gbk → ../results/genome_fastas/NZ_CP124347.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP035720.1.gbk → ../results/genome_fastas/NZ_CP035720.1.fasta\n",
      "Converting: ../data/gbk/NZ_LS992192.1.gbk → ../results/genome_fastas/NZ_LS992192.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP076687.1.gbk → ../results/genome_fastas/NZ_CP076687.1.fasta\n",
      "Converting: ../data/gbk/NC_022648.1.gbk → ../results/genome_fastas/NC_022648.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124326.1.gbk → ../results/genome_fastas/NZ_CP124326.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124436.1.gbk → ../results/genome_fastas/NZ_CP124436.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP022064.1.gbk → ../results/genome_fastas/NZ_AP022064.1.fasta\n",
      "Converting: ../data/gbk/NZ_JAPVBX010000001.1.gbk → ../results/genome_fastas/NZ_JAPVBX010000001.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP051615.1.gbk → ../results/genome_fastas/NZ_CP051615.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP134384.1.gbk → ../results/genome_fastas/NZ_CP134384.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP015159.1.gbk → ../results/genome_fastas/NZ_CP015159.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107120.1.gbk → ../results/genome_fastas/NZ_CP107120.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124415.1.gbk → ../results/genome_fastas/NZ_CP124415.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124508.1.gbk → ../results/genome_fastas/NZ_CP124508.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP014495.1.gbk → ../results/genome_fastas/NZ_CP014495.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP047710.1.gbk → ../results/genome_fastas/NZ_CP047710.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP028483.1.gbk → ../results/genome_fastas/NZ_CP028483.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP036245.1.gbk → ../results/genome_fastas/NZ_CP036245.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124511.1.gbk → ../results/genome_fastas/NZ_CP124511.1.fasta\n",
      "Converting: ../data/gbk/NZ_LR130545.1.gbk → ../results/genome_fastas/NZ_LR130545.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP018784.1.gbk → ../results/genome_fastas/NZ_AP018784.1.fasta\n",
      "Converting: ../data/gbk/NZ_LS992168.1.gbk → ../results/genome_fastas/NZ_LS992168.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP080119.1.gbk → ../results/genome_fastas/NZ_CP080119.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124359.1.gbk → ../results/genome_fastas/NZ_CP124359.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP022362.1.gbk → ../results/genome_fastas/NZ_AP022362.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP015138.1.gbk → ../results/genome_fastas/NZ_CP015138.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107143.1.gbk → ../results/genome_fastas/NZ_CP107143.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP029420.1.gbk → ../results/genome_fastas/NZ_CP029420.1.fasta\n",
      "Converting: ../data/gbk/NZ_LS999560.1.gbk → ../results/genome_fastas/NZ_LS999560.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP018983.1.gbk → ../results/genome_fastas/NZ_CP018983.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124513.1.gbk → ../results/genome_fastas/NZ_CP124513.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP019020.1.gbk → ../results/genome_fastas/NZ_CP019020.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP022549.1.gbk → ../results/genome_fastas/NZ_AP022549.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP051609.1.gbk → ../results/genome_fastas/NZ_CP051609.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124492.1.gbk → ../results/genome_fastas/NZ_CP124492.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP059130.1.gbk → ../results/genome_fastas/NZ_CP059130.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107122.1.gbk → ../results/genome_fastas/NZ_CP107122.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP097360.1.gbk → ../results/genome_fastas/NZ_CP097360.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP014497.1.gbk → ../results/genome_fastas/NZ_CP014497.1.fasta\n",
      "Converting: ../data/gbk/NZ_LR890693.1.gbk → ../results/genome_fastas/NZ_LR890693.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP021689.1.gbk → ../results/genome_fastas/NZ_CP021689.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP019000.1.gbk → ../results/genome_fastas/NZ_CP019000.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP023849.1.gbk → ../results/genome_fastas/NZ_CP023849.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107126.1.gbk → ../results/genome_fastas/NZ_CP107126.1.fasta\n",
      "Converting: ../data/gbk/NZ_OW849064.1.gbk → ../results/genome_fastas/NZ_OW849064.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP021179.1.gbk → ../results/genome_fastas/NZ_CP021179.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP069488.1.gbk → ../results/genome_fastas/NZ_CP069488.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103465.1.gbk → ../results/genome_fastas/NZ_CP103465.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107147.1.gbk → ../results/genome_fastas/NZ_CP107147.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP128907.1.gbk → ../results/genome_fastas/NZ_CP128907.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124517.1.gbk → ../results/genome_fastas/NZ_CP124517.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP080120.1.gbk → ../results/genome_fastas/NZ_CP080120.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107178.1.gbk → ../results/genome_fastas/NZ_CP107178.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP014488.1.gbk → ../results/genome_fastas/NZ_CP014488.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP024720.1.gbk → ../results/genome_fastas/NZ_CP024720.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124515.1.gbk → ../results/genome_fastas/NZ_CP124515.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP095540.1.gbk → ../results/genome_fastas/NZ_CP095540.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP104848.1.gbk → ../results/genome_fastas/NZ_CP104848.1.fasta\n",
      "Converting: ../data/gbk/NZ_OX637964.1.gbk → ../results/genome_fastas/NZ_OX637964.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103742.1.gbk → ../results/genome_fastas/NZ_CP103742.1.fasta\n",
      "Converting: ../data/gbk/NZ_OX637959.1.gbk → ../results/genome_fastas/NZ_OX637959.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP063774.1.gbk → ../results/genome_fastas/NZ_CP063774.1.fasta\n",
      "Converting: ../data/gbk/NZ_OX030701.1.gbk → ../results/genome_fastas/NZ_OX030701.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124498.1.gbk → ../results/genome_fastas/NZ_CP124498.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124420.1.gbk → ../results/genome_fastas/NZ_CP124420.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103592.1.gbk → ../results/genome_fastas/NZ_CP103592.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP020116.1.gbk → ../results/genome_fastas/NZ_CP020116.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124500.1.gbk → ../results/genome_fastas/NZ_CP124500.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107128.1.gbk → ../results/genome_fastas/NZ_CP107128.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124481.1.gbk → ../results/genome_fastas/NZ_CP124481.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124351.1.gbk → ../results/genome_fastas/NZ_CP124351.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP022730.1.gbk → ../results/genome_fastas/NZ_CP022730.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP023826.1.gbk → ../results/genome_fastas/NZ_CP023826.1.fasta\n",
      "Converting: ../data/gbk/NZ_OY019088.1.gbk → ../results/genome_fastas/NZ_OY019088.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107174.1.gbk → ../results/genome_fastas/NZ_CP107174.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103755.1.gbk → ../results/genome_fastas/NZ_CP103755.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103710.1.gbk → ../results/genome_fastas/NZ_CP103710.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP013835.1.gbk → ../results/genome_fastas/NZ_CP013835.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP049085.2.gbk → ../results/genome_fastas/NZ_CP049085.2.fasta\n",
      "Converting: ../data/gbk/NZ_CP056114.1.gbk → ../results/genome_fastas/NZ_CP056114.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124353.1.gbk → ../results/genome_fastas/NZ_CP124353.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP019015.1.gbk → ../results/genome_fastas/NZ_CP019015.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124483.1.gbk → ../results/genome_fastas/NZ_CP124483.1.fasta\n",
      "Converting: ../data/gbk/NZ_HG941718.1.gbk → ../results/genome_fastas/NZ_HG941718.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124502.1.gbk → ../results/genome_fastas/NZ_CP124502.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP104846.1.gbk → ../results/genome_fastas/NZ_CP104846.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP102061.1.gbk → ../results/genome_fastas/NZ_CP102061.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP076693.1.gbk → ../results/genome_fastas/NZ_CP076693.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124332.1.gbk → ../results/genome_fastas/NZ_CP124332.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP059281.1.gbk → ../results/genome_fastas/NZ_CP059281.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124467.1.gbk → ../results/genome_fastas/NZ_CP124467.1.fasta\n",
      "Converting: ../data/gbk/NZ_OW848785.1.gbk → ../results/genome_fastas/NZ_OW848785.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107117.1.gbk → ../results/genome_fastas/NZ_CP107117.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP102947.1.gbk → ../results/genome_fastas/NZ_CP102947.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP128916.1.gbk → ../results/genome_fastas/NZ_CP128916.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP076697.1.gbk → ../results/genome_fastas/NZ_CP076697.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124506.1.gbk → ../results/genome_fastas/NZ_CP124506.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP019008.1.gbk → ../results/genome_fastas/NZ_CP019008.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103508.1.gbk → ../results/genome_fastas/NZ_CP103508.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP013831.1.gbk → ../results/genome_fastas/NZ_CP013831.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP041337.1.gbk → ../results/genome_fastas/NZ_CP041337.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107137.1.gbk → ../results/genome_fastas/NZ_CP107137.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107172.1.gbk → ../results/genome_fastas/NZ_CP107172.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124487.1.gbk → ../results/genome_fastas/NZ_CP124487.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP051659.1.gbk → ../results/genome_fastas/NZ_CP051659.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP032265.1.gbk → ../results/genome_fastas/NZ_CP032265.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124357.1.gbk → ../results/genome_fastas/NZ_CP124357.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP080117.1.gbk → ../results/genome_fastas/NZ_CP080117.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP070593.1.gbk → ../results/genome_fastas/NZ_CP070593.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP024717.1.gbk → ../results/genome_fastas/NZ_CP024717.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124355.1.gbk → ../results/genome_fastas/NZ_CP124355.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP035476.1.gbk → ../results/genome_fastas/NZ_CP035476.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP029576.1.gbk → ../results/genome_fastas/NZ_CP029576.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP018970.1.gbk → ../results/genome_fastas/NZ_CP018970.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124400.1.gbk → ../results/genome_fastas/NZ_CP124400.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP098203.1.gbk → ../results/genome_fastas/NZ_CP098203.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124424.1.gbk → ../results/genome_fastas/NZ_CP124424.1.fasta\n",
      "Converting: ../data/gbk/NZ_LS992180.1.gbk → ../results/genome_fastas/NZ_LS992180.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124334.1.gbk → ../results/genome_fastas/NZ_CP124334.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP026788.1.gbk → ../results/genome_fastas/NZ_AP026788.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP097370.1.gbk → ../results/genome_fastas/NZ_CP097370.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP030337.1.gbk → ../results/genome_fastas/NZ_CP030337.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP051661.1.gbk → ../results/genome_fastas/NZ_CP051661.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124392.1.gbk → ../results/genome_fastas/NZ_CP124392.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP014522.1.gbk → ../results/genome_fastas/NZ_CP014522.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124442.1.gbk → ../results/genome_fastas/NZ_CP124442.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP018953.1.gbk → ../results/genome_fastas/NZ_CP018953.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP023844.1.gbk → ../results/genome_fastas/NZ_CP023844.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107153.1.gbk → ../results/genome_fastas/NZ_CP107153.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP032201.1.gbk → ../results/genome_fastas/NZ_CP032201.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP021998.1.gbk → ../results/genome_fastas/NZ_AP021998.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124374.1.gbk → ../results/genome_fastas/NZ_CP124374.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP018991.1.gbk → ../results/genome_fastas/NZ_CP018991.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107151.1.gbk → ../results/genome_fastas/NZ_CP107151.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107114.1.gbk → ../results/genome_fastas/NZ_CP107114.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP069583.1.gbk → ../results/genome_fastas/NZ_CP069583.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124349.1.gbk → ../results/genome_fastas/NZ_CP124349.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP035516.1.gbk → ../results/genome_fastas/NZ_CP035516.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP026794.1.gbk → ../results/genome_fastas/NZ_AP026794.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP054219.1.gbk → ../results/genome_fastas/NZ_CP054219.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107188.1.gbk → ../results/genome_fastas/NZ_CP107188.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124405.1.gbk → ../results/genome_fastas/NZ_CP124405.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP076689.1.gbk → ../results/genome_fastas/NZ_CP076689.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124328.1.gbk → ../results/genome_fastas/NZ_CP124328.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP051663.1.gbk → ../results/genome_fastas/NZ_CP051663.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP015069.1.gbk → ../results/genome_fastas/NZ_CP015069.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP021935.1.gbk → ../results/genome_fastas/NZ_CP021935.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107134.1.gbk → ../results/genome_fastas/NZ_CP107134.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103672.1.gbk → ../results/genome_fastas/NZ_CP103672.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP019012.1.gbk → ../results/genome_fastas/NZ_CP019012.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP035477.1.gbk → ../results/genome_fastas/NZ_CP035477.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP120633.1.gbk → ../results/genome_fastas/NZ_CP120633.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP103557.1.gbk → ../results/genome_fastas/NZ_CP103557.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP013658.1.gbk → ../results/genome_fastas/NZ_CP013658.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP115324.1.gbk → ../results/genome_fastas/NZ_CP115324.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124460.1.gbk → ../results/genome_fastas/NZ_CP124460.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP107155.1.gbk → ../results/genome_fastas/NZ_CP107155.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP022032.1.gbk → ../results/genome_fastas/NZ_AP022032.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP015076.1.gbk → ../results/genome_fastas/NZ_CP015076.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP082775.1.gbk → ../results/genome_fastas/NZ_CP082775.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124337.1.gbk → ../results/genome_fastas/NZ_CP124337.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP124372.1.gbk → ../results/genome_fastas/NZ_CP124372.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP059279.1.gbk → ../results/genome_fastas/NZ_CP059279.1.fasta\n",
      "Converting: ../data/gbk/NZ_AP022171.1.gbk → ../results/genome_fastas/NZ_AP022171.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP098219.1.gbk → ../results/genome_fastas/NZ_CP098219.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP096110.1.gbk → ../results/genome_fastas/NZ_CP096110.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP080116.1.gbk → ../results/genome_fastas/NZ_CP080116.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP015074.2.gbk → ../results/genome_fastas/NZ_CP015074.2.fasta\n",
      "Converting: ../data/gbk/NZ_AP022525.1.gbk → ../results/genome_fastas/NZ_AP022525.1.fasta\n",
      "Converting: ../data/gbk/NZ_OW968277.1.gbk → ../results/genome_fastas/NZ_OW968277.1.fasta\n",
      "Converting: ../data/gbk/NZ_CP014316.1.gbk → ../results/genome_fastas/NZ_CP014316.1.fasta\n"
     ]
    }
   ],
   "source": [
    "# make sure that all genome fasta files exist\n",
    "convert_gbk_fasta(gbk_folder = \"../data/gbk\", fasta_folder = \"../results/genome_fastas\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55e9aaf1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::0.077*1.09] collected minimizers\n",
      "[M::mm_idx_gen::0.085*1.28] sorted minimizers\n",
      "[M::main::0.085*1.28] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.091*1.26] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.095*1.25] distinct minimizers: 504461 (98.70% are singletons); average occurrences: 1.024; average spacing: 9.992; total length: 5163545\n",
      "[M::worker_pipeline::0.096*1.25] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP124455.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP124455.1_segment_0.fasta\n",
      "[M::main] Real time: 0.102 sec; CPU: 0.125 sec; Peak RSS: 0.045 GB\n",
      "[M::mm_idx_gen::0.064*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.24] sorted minimizers\n",
      "[M::main::0.071*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 496890 (99.14% are singletons); average occurrences: 1.021; average spacing: 9.992; total length: 5071232\n",
      "[M::worker_pipeline::0.080*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP107122.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP107122.1_segment_0.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.104 sec; Peak RSS: 0.045 GB\n",
      "[M::mm_idx_gen::0.061*1.05] collected minimizers\n",
      "[M::mm_idx_gen::0.069*1.27] sorted minimizers\n",
      "[M::main::0.069*1.27] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.073*1.25] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.078*1.24] distinct minimizers: 488846 (99.19% are singletons); average occurrences: 1.016; average spacing: 9.993; total length: 4965000\n",
      "[M::worker_pipeline::0.078*1.24] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP128947.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP128947.1_segment_0.fasta\n",
      "[M::main] Real time: 0.085 sec; CPU: 0.103 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.066*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.24] sorted minimizers\n",
      "[M::main::0.075*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.22] distinct minimizers: 512762 (99.05% are singletons); average occurrences: 1.018; average spacing: 9.995; total length: 5216726\n",
      "[M::worker_pipeline::0.084*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP059130.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP059130.1_segment_1.fasta\n",
      "[M::main] Real time: 0.092 sec; CPU: 0.110 sec; Peak RSS: 0.046 GB\n",
      "[M::mm_idx_gen::0.062*1.05] collected minimizers\n",
      "[M::mm_idx_gen::0.070*1.26] sorted minimizers\n",
      "[M::main::0.070*1.26] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.23] distinct minimizers: 489014 (99.15% are singletons); average occurrences: 1.019; average spacing: 9.993; total length: 4978191\n",
      "[M::worker_pipeline::0.080*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP124460.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP124460.1_segment_0.fasta\n",
      "[M::main] Real time: 0.086 sec; CPU: 0.104 sec; Peak RSS: 0.042 GB\n",
      "[M::mm_idx_gen::0.062*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.070*1.25] sorted minimizers\n",
      "[M::main::0.070*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 495762 (98.85% are singletons); average occurrences: 1.022; average spacing: 9.992; total length: 5061196\n",
      "[M::worker_pipeline::0.081*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_LR890693.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_LR890693.1_segment_0.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.105 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.061*1.05] collected minimizers\n",
      "[M::mm_idx_gen::0.069*1.26] sorted minimizers\n",
      "[M::main::0.069*1.26] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.073*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.077*1.23] distinct minimizers: 492759 (99.28% are singletons); average occurrences: 1.018; average spacing: 9.994; total length: 5010884\n",
      "[M::worker_pipeline::0.078*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP010876.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP010876.1_segment_0.fasta\n",
      "[M::main] Real time: 0.084 sec; CPU: 0.102 sec; Peak RSS: 0.041 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.25] sorted minimizers\n",
      "[M::main::0.071*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.22] distinct minimizers: 504923 (99.18% are singletons); average occurrences: 1.017; average spacing: 9.993; total length: 5129468\n",
      "[M::worker_pipeline::0.081*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP023826.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP023826.1_segment_0.fasta\n",
      "[M::main] Real time: 0.088 sec; CPU: 0.105 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.065*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.28] sorted minimizers\n",
      "[M::main::0.075*1.28] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.26] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.24] distinct minimizers: 489613 (99.23% are singletons); average occurrences: 1.018; average spacing: 9.993; total length: 4978342\n",
      "[M::worker_pipeline::0.085*1.24] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP021454.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP021454.1_segment_0.fasta\n",
      "[M::main] Real time: 0.093 sec; CPU: 0.113 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.065*1.06] collected minimizers\n",
      "[M::mm_idx_gen::0.073*1.27] sorted minimizers\n",
      "[M::main::0.073*1.27] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.084*1.23] distinct minimizers: 472327 (97.85% are singletons); average occurrences: 1.037; average spacing: 9.995; total length: 4893409\n",
      "[M::worker_pipeline::0.084*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP124467.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP124467.1_segment_0.fasta\n",
      "[M::main] Real time: 0.092 sec; CPU: 0.111 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.066*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.24] sorted minimizers\n",
      "[M::main::0.075*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.21] distinct minimizers: 488238 (99.19% are singletons); average occurrences: 1.019; average spacing: 9.994; total length: 4974345\n",
      "[M::worker_pipeline::0.084*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP124429.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP124429.1_segment_0.fasta\n",
      "[M::main] Real time: 0.092 sec; CPU: 0.110 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.064*1.05] collected minimizers\n",
      "[M::mm_idx_gen::0.072*1.26] sorted minimizers\n",
      "[M::main::0.073*1.26] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.23] distinct minimizers: 495762 (98.85% are singletons); average occurrences: 1.022; average spacing: 9.992; total length: 5061196\n",
      "[M::worker_pipeline::0.083*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_LR890693.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_LR890693.1_segment_1.fasta\n",
      "[M::main] Real time: 0.090 sec; CPU: 0.109 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.065*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.073*1.24] sorted minimizers\n",
      "[M::main::0.073*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.21] distinct minimizers: 506371 (99.21% are singletons); average occurrences: 1.016; average spacing: 9.993; total length: 5141314\n",
      "[M::worker_pipeline::0.083*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP021935.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP021935.1_segment_0.fasta\n",
      "[M::main] Real time: 0.090 sec; CPU: 0.107 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.068*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.076*1.24] sorted minimizers\n",
      "[M::main::0.076*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.086*1.21] distinct minimizers: 509570 (97.84% are singletons); average occurrences: 1.033; average spacing: 9.996; total length: 5259570\n",
      "[M::worker_pipeline::0.087*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP107114.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP107114.1_segment_0.fasta\n",
      "[M::main] Real time: 0.094 sec; CPU: 0.112 sec; Peak RSS: 0.054 GB\n",
      "[M::mm_idx_gen::0.065*1.02] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.24] sorted minimizers\n",
      "[M::main::0.074*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.21] distinct minimizers: 500234 (98.88% are singletons); average occurrences: 1.019; average spacing: 9.995; total length: 5096686\n",
      "[M::worker_pipeline::0.083*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP107172.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP107172.1_segment_0.fasta\n",
      "[M::main] Real time: 0.090 sec; CPU: 0.107 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.25] sorted minimizers\n",
      "[M::main::0.071*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 495194 (98.93% are singletons); average occurrences: 1.020; average spacing: 9.992; total length: 5048620\n",
      "[M::worker_pipeline::0.080*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP051659.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP051659.1_segment_0.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.105 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.066*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.24] sorted minimizers\n",
      "[M::main::0.074*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.22] distinct minimizers: 510090 (98.23% are singletons); average occurrences: 1.026; average spacing: 9.994; total length: 5228864\n",
      "[M::worker_pipeline::0.083*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP018979.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP018979.1_segment_0.fasta\n",
      "[M::main] Real time: 0.089 sec; CPU: 0.107 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.060*1.06] collected minimizers\n",
      "[M::mm_idx_gen::0.068*1.28] sorted minimizers\n",
      "[M::main::0.068*1.28] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.073*1.26] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.077*1.25] distinct minimizers: 496890 (99.14% are singletons); average occurrences: 1.021; average spacing: 9.992; total length: 5071232\n",
      "[M::worker_pipeline::0.077*1.25] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP107122.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP107122.1_segment_1.fasta\n",
      "[M::main] Real time: 0.084 sec; CPU: 0.103 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.063*1.06] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.26] sorted minimizers\n",
      "[M::main::0.071*1.26] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.23] distinct minimizers: 512762 (99.05% are singletons); average occurrences: 1.018; average spacing: 9.995; total length: 5216726\n",
      "[M::worker_pipeline::0.081*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP059130.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP059130.1_segment_0.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.106 sec; Peak RSS: 0.048 GB\n",
      "[M::mm_idx_gen::0.061*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.069*1.26] sorted minimizers\n",
      "[M::main::0.069*1.26] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.073*1.25] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.077*1.24] distinct minimizers: 493022 (99.23% are singletons); average occurrences: 1.017; average spacing: 9.993; total length: 5012084\n",
      "[M::worker_pipeline::0.078*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP021179.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP021179.1_segment_0.fasta\n",
      "[M::main] Real time: 0.085 sec; CPU: 0.103 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.066*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.25] sorted minimizers\n",
      "[M::main::0.074*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.22] distinct minimizers: 514601 (98.63% are singletons); average occurrences: 1.023; average spacing: 9.992; total length: 5259387\n",
      "[M::worker_pipeline::0.084*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP018970.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP018970.1_segment_0.fasta\n",
      "[M::main] Real time: 0.090 sec; CPU: 0.109 sec; Peak RSS: 0.053 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.070*1.25] sorted minimizers\n",
      "[M::main::0.070*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.079*1.22] distinct minimizers: 496945 (98.97% are singletons); average occurrences: 1.021; average spacing: 9.993; total length: 5070006\n",
      "[M::worker_pipeline::0.080*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_OX030701.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_OX030701.1_segment_0.fasta\n",
      "[M::main] Real time: 0.086 sec; CPU: 0.104 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.24] sorted minimizers\n",
      "[M::main::0.071*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 507513 (98.75% are singletons); average occurrences: 1.021; average spacing: 9.994; total length: 5181038\n",
      "[M::worker_pipeline::0.080*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP076693.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP076693.1_segment_0.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.104 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.070*1.25] sorted minimizers\n",
      "[M::main::0.070*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.075*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 502386 (99.09% are singletons); average occurrences: 1.022; average spacing: 9.992; total length: 5128182\n",
      "[M::worker_pipeline::0.080*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP107162.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP107162.1_segment_0.fasta\n",
      "[M::main] Real time: 0.088 sec; CPU: 0.105 sec; Peak RSS: 0.042 GB\n",
      "[M::mm_idx_gen::0.061*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.069*1.25] sorted minimizers\n",
      "[M::main::0.069*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.073*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.077*1.23] distinct minimizers: 491236 (98.56% are singletons); average occurrences: 1.026; average spacing: 9.994; total length: 5038168\n",
      "[M::worker_pipeline::0.078*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP022326.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP022326.1_segment_0.fasta\n",
      "[M::main] Real time: 0.084 sec; CPU: 0.102 sec; Peak RSS: 0.042 GB\n",
      "[M::mm_idx_gen::0.063*1.05] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.26] sorted minimizers\n",
      "[M::main::0.071*1.26] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.23] distinct minimizers: 508685 (98.70% are singletons); average occurrences: 1.020; average spacing: 9.995; total length: 5187769\n",
      "[M::worker_pipeline::0.081*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP036245.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP036245.1_segment_0.fasta\n",
      "[M::main] Real time: 0.088 sec; CPU: 0.106 sec; Peak RSS: 0.045 GB\n",
      "[M::mm_idx_gen::0.060*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.068*1.26] sorted minimizers\n",
      "[M::main::0.068*1.26] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.072*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.076*1.23] distinct minimizers: 484797 (99.39% are singletons); average occurrences: 1.015; average spacing: 9.991; total length: 4916220\n",
      "[M::worker_pipeline::0.077*1.23] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP128907.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP128907.1_segment_0.fasta\n",
      "[M::main] Real time: 0.083 sec; CPU: 0.101 sec; Peak RSS: 0.041 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.25] sorted minimizers\n",
      "[M::main::0.071*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 508265 (98.56% are singletons); average occurrences: 1.024; average spacing: 9.993; total length: 5198889\n",
      "[M::worker_pipeline::0.080*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP026794.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP026794.1_segment_1.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.104 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.067*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.23] sorted minimizers\n",
      "[M::main::0.075*1.23] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.20] distinct minimizers: 508015 (98.55% are singletons); average occurrences: 1.023; average spacing: 9.994; total length: 5194327\n",
      "[M::worker_pipeline::0.086*1.20] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP026788.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP026788.1_segment_1.fasta\n",
      "[M::main] Real time: 0.093 sec; CPU: 0.110 sec; Peak RSS: 0.046 GB\n",
      "[M::mm_idx_gen::0.062*1.05] collected minimizers\n",
      "[M::mm_idx_gen::0.070*1.27] sorted minimizers\n",
      "[M::main::0.070*1.27] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.074*1.26] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.078*1.24] distinct minimizers: 481518 (98.93% are singletons); average occurrences: 1.017; average spacing: 9.991; total length: 4893245\n",
      "[M::worker_pipeline::0.079*1.24] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP134384.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP134384.1_segment_0.fasta\n",
      "[M::main] Real time: 0.086 sec; CPU: 0.105 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.064*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.073*1.25] sorted minimizers\n",
      "[M::main::0.073*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.22] distinct minimizers: 492261 (98.84% are singletons); average occurrences: 1.021; average spacing: 9.993; total length: 5023946\n",
      "[M::worker_pipeline::0.082*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP051661.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP051661.1_segment_0.fasta\n",
      "[M::main] Real time: 0.090 sec; CPU: 0.108 sec; Peak RSS: 0.041 GB\n",
      "[M::mm_idx_gen::0.063*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.072*1.24] sorted minimizers\n",
      "[M::main::0.072*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.22] distinct minimizers: 487862 (99.17% are singletons); average occurrences: 1.021; average spacing: 9.993; total length: 4977635\n",
      "[M::worker_pipeline::0.081*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP124424.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP124424.1_segment_0.fasta\n",
      "[M::main] Real time: 0.088 sec; CPU: 0.106 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.114*0.66] collected minimizers\n",
      "[M::mm_idx_gen::0.124*0.81] sorted minimizers\n",
      "[M::main::0.124*0.81] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.129*0.82] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.133*0.82] distinct minimizers: 489986 (99.20% are singletons); average occurrences: 1.019; average spacing: 9.994; total length: 4989783\n",
      "[M::worker_pipeline::0.133*0.83] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP030337.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP030337.1_segment_0.fasta\n",
      "[M::main] Real time: 0.141 sec; CPU: 0.117 sec; Peak RSS: 0.041 GB\n",
      "[M::mm_idx_gen::0.065*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.072*1.23] sorted minimizers\n",
      "[M::main::0.072*1.23] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.21] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.082*1.20] distinct minimizers: 489584 (99.20% are singletons); average occurrences: 1.021; average spacing: 9.993; total length: 4993064\n",
      "[M::worker_pipeline::0.082*1.20] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP124410.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP124410.1_segment_0.fasta\n",
      "[M::main] Real time: 0.091 sec; CPU: 0.107 sec; Peak RSS: 0.042 GB\n",
      "[M::mm_idx_gen::0.066*1.02] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.23] sorted minimizers\n",
      "[M::main::0.074*1.23] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.079*1.21] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.20] distinct minimizers: 495273 (99.19% are singletons); average occurrences: 1.018; average spacing: 9.994; total length: 5040845\n",
      "[M::worker_pipeline::0.084*1.20] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP022044.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP022044.1_segment_0.fasta\n",
      "[M::main] Real time: 0.092 sec; CPU: 0.108 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.063*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.23] sorted minimizers\n",
      "[M::main::0.071*1.23] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.21] distinct minimizers: 482985 (99.20% are singletons); average occurrences: 1.018; average spacing: 9.994; total length: 4911496\n",
      "[M::worker_pipeline::0.081*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP019012.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP019012.1_segment_0.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.104 sec; Peak RSS: 0.042 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.25] sorted minimizers\n",
      "[M::main::0.071*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.22] distinct minimizers: 495880 (99.41% are singletons); average occurrences: 1.012; average spacing: 9.991; total length: 5014829\n",
      "[M::worker_pipeline::0.081*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP133923.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP133923.1_segment_0.fasta\n",
      "[M::main] Real time: 0.088 sec; CPU: 0.106 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.065*1.06] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.27] sorted minimizers\n",
      "[M::main::0.074*1.27] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.26] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.24] distinct minimizers: 508265 (98.56% are singletons); average occurrences: 1.024; average spacing: 9.993; total length: 5198889\n",
      "[M::worker_pipeline::0.084*1.24] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP026794.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP026794.1_segment_0.fasta\n",
      "[M::main] Real time: 0.090 sec; CPU: 0.110 sec; Peak RSS: 0.048 GB\n",
      "[M::mm_idx_gen::0.070*0.99] collected minimizers\n",
      "[M::mm_idx_gen::0.078*1.19] sorted minimizers\n",
      "[M::main::0.078*1.19] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.083*1.18] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.088*1.17] distinct minimizers: 490008 (98.63% are singletons); average occurrences: 1.026; average spacing: 9.994; total length: 5022408\n",
      "[M::worker_pipeline::0.088*1.16] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP027534.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP027534.1_segment_0.fasta\n",
      "[M::main] Real time: 0.095 sec; CPU: 0.109 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.062*1.07] collected minimizers\n",
      "[M::mm_idx_gen::0.072*1.29] sorted minimizers\n",
      "[M::main::0.072*1.29] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.27] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.26] distinct minimizers: 492261 (98.84% are singletons); average occurrences: 1.021; average spacing: 9.993; total length: 5023946\n",
      "[M::worker_pipeline::0.081*1.26] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP051661.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP051661.1_segment_1.fasta\n",
      "[M::main] Real time: 0.088 sec; CPU: 0.108 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.068*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.077*1.23] sorted minimizers\n",
      "[M::main::0.077*1.23] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.082*1.21] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.087*1.20] distinct minimizers: 508015 (98.55% are singletons); average occurrences: 1.023; average spacing: 9.994; total length: 5194327\n",
      "[M::worker_pipeline::0.087*1.20] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP026788.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP026788.1_segment_0.fasta\n",
      "[M::main] Real time: 0.094 sec; CPU: 0.112 sec; Peak RSS: 0.046 GB\n",
      "[M::mm_idx_gen::0.064*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.073*1.23] sorted minimizers\n",
      "[M::main::0.073*1.23] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.077*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.081*1.21] distinct minimizers: 487813 (99.53% are singletons); average occurrences: 1.011; average spacing: 9.990; total length: 4926149\n",
      "[M::worker_pipeline::0.082*1.20] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP019008.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP019008.1_segment_0.fasta\n",
      "[M::main] Real time: 0.089 sec; CPU: 0.105 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.064*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.071*1.25] sorted minimizers\n",
      "[M::main::0.071*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 491236 (98.56% are singletons); average occurrences: 1.026; average spacing: 9.994; total length: 5038168\n",
      "[M::worker_pipeline::0.081*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP022326.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP022326.1_segment_1.fasta\n",
      "[M::main] Real time: 0.087 sec; CPU: 0.105 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.067*1.03] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.24] sorted minimizers\n",
      "[M::main::0.075*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.081*1.22] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.21] distinct minimizers: 505113 (99.01% are singletons); average occurrences: 1.020; average spacing: 9.993; total length: 5151251\n",
      "[M::worker_pipeline::0.086*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP124374.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP124374.1_segment_0.fasta\n",
      "[M::main] Real time: 0.094 sec; CPU: 0.111 sec; Peak RSS: 0.045 GB\n",
      "[M::mm_idx_gen::0.070*1.01] collected minimizers\n",
      "[M::mm_idx_gen::0.079*1.20] sorted minimizers\n",
      "[M::main::0.079*1.20] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.084*1.19] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.088*1.18] distinct minimizers: 501904 (99.12% are singletons); average occurrences: 1.021; average spacing: 9.992; total length: 5120420\n",
      "[M::worker_pipeline::0.089*1.18] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP107151.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP107151.1_segment_0.fasta\n",
      "[M::main] Real time: 0.095 sec; CPU: 0.111 sec; Peak RSS: 0.045 GB\n",
      "[M::mm_idx_gen::0.065*1.06] collected minimizers\n",
      "[M::mm_idx_gen::0.075*1.24] sorted minimizers\n",
      "[M::main::0.075*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.080*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.085*1.21] distinct minimizers: 491168 (98.88% are singletons); average occurrences: 1.020; average spacing: 9.992; total length: 5007059\n",
      "[M::worker_pipeline::0.086*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP051663.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP051663.1_segment_0.fasta\n",
      "[M::main] Real time: 0.092 sec; CPU: 0.109 sec; Peak RSS: 0.044 GB\n",
      "[M::mm_idx_gen::0.065*1.05] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.25] sorted minimizers\n",
      "[M::main::0.074*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.24] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.22] distinct minimizers: 502143 (99.17% are singletons); average occurrences: 1.016; average spacing: 9.994; total length: 5099988\n",
      "[M::worker_pipeline::0.083*1.22] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP019015.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP019015.1_segment_0.fasta\n",
      "[M::main] Real time: 0.089 sec; CPU: 0.108 sec; Peak RSS: 0.047 GB\n",
      "[M::mm_idx_gen::0.066*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.074*1.24] sorted minimizers\n",
      "[M::main::0.074*1.24] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.078*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.083*1.21] distinct minimizers: 510387 (98.93% are singletons); average occurrences: 1.022; average spacing: 9.995; total length: 5211978\n",
      "[M::worker_pipeline::0.084*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP103562.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_CP103562.1_segment_0.fasta\n",
      "[M::main] Real time: 0.090 sec; CPU: 0.108 sec; Peak RSS: 0.046 GB\n",
      "[M::mm_idx_gen::0.060*1.06] collected minimizers\n",
      "[M::mm_idx_gen::0.068*1.28] sorted minimizers\n",
      "[M::main::0.068*1.28] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.072*1.26] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.076*1.25] distinct minimizers: 495273 (99.19% are singletons); average occurrences: 1.018; average spacing: 9.994; total length: 5040845\n",
      "[M::worker_pipeline::0.077*1.25] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP022044.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus1/NZ_AP022044.1_segment_1.fasta\n",
      "[M::main] Real time: 0.083 sec; CPU: 0.102 sec; Peak RSS: 0.042 GB\n",
      "[M::mm_idx_gen::0.103*0.70] collected minimizers\n",
      "[M::mm_idx_gen::0.110*0.85] sorted minimizers\n",
      "[M::main::0.110*0.85] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.116*0.86] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.120*0.86] distinct minimizers: 482392 (99.13% are singletons); average occurrences: 1.019; average spacing: 9.991; total length: 4909078\n",
      "[M::worker_pipeline::0.124*0.86] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_CP098219.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2/NZ_CP098219.1_segment_0.fasta\n",
      "[M::main] Real time: 0.132 sec; CPU: 0.115 sec; Peak RSS: 0.043 GB\n",
      "[M::mm_idx_gen::0.063*1.04] collected minimizers\n",
      "[M::mm_idx_gen::0.072*1.25] sorted minimizers\n",
      "[M::main::0.072*1.25] loaded/built the index for 1 target sequence(s)\n",
      "[M::mm_mapopt_update::0.076*1.23] mid_occ = 50\n",
      "[M::mm_idx_stat] kmer size: 19; skip: 19; is_hpc: 0; #seq: 1\n",
      "[M::mm_idx_stat::0.080*1.22] distinct minimizers: 496276 (98.84% are singletons); average occurrences: 1.022; average spacing: 9.988; total length: 5066983\n",
      "[M::worker_pipeline::0.081*1.21] mapped 1 sequences\n",
      "[M::main] Version: 2.30-r1287\n",
      "[M::main] CMD: minimap2 -x asm5 -N 50 -p 0.9 -k 19 --eqx ../results/genome_fastas/NZ_AP022171.1.fasta ../results/atb_lookup/CIRMBUYJFK_f__CWCCKOQCWZ_r/consensus2/NZ_AP022171.1_segment_0.fasta\n",
      "[M::main] Real time: 0.089 sec; CPU: 0.106 sec; Peak RSS: 0.044 GB\n"
     ]
    }
   ],
   "source": [
    "insertions_df = find_insertion_hits_own_genome(genome_root = \"../results/genome_fastas\", insertions_seq_dir=\"../results/atb_lookup\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "377b8be5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction_name</th>\n",
       "      <th>consensus</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>insertion_path</th>\n",
       "      <th>insertion_length</th>\n",
       "      <th>segment</th>\n",
       "      <th>hits_in_genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP026794.1</td>\n",
       "      <td>16556873922195442803|+|</td>\n",
       "      <td>1661</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP026788.1</td>\n",
       "      <td>16556873922195442803|+|</td>\n",
       "      <td>1661</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP124467.1</td>\n",
       "      <td>185719230501067033|+|</td>\n",
       "      <td>4204</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP134384.1</td>\n",
       "      <td>185719230501067033|+|</td>\n",
       "      <td>4203</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP133923.1</td>\n",
       "      <td>185719230501067033|+|</td>\n",
       "      <td>4204</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP019008.1</td>\n",
       "      <td>185719230501067033|+|</td>\n",
       "      <td>4204</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP021935.1</td>\n",
       "      <td>517157241956952768|+|</td>\n",
       "      <td>1248</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP107114.1</td>\n",
       "      <td>517157241956952768|+|</td>\n",
       "      <td>1248</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP107172.1</td>\n",
       "      <td>517157241956952768|+|</td>\n",
       "      <td>1248</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP059130.1</td>\n",
       "      <td>517157241956952768|+|</td>\n",
       "      <td>1253</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP021179.1</td>\n",
       "      <td>517157241956952768|+|</td>\n",
       "      <td>1248</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP124429.1</td>\n",
       "      <td>517157241956952768|-|</td>\n",
       "      <td>1248</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP018979.1</td>\n",
       "      <td>517157241956952768|-|</td>\n",
       "      <td>1248</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP051659.1</td>\n",
       "      <td>6932368721519041699|+|10145330714100251892</td>\n",
       "      <td>760</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP051661.1</td>\n",
       "      <td>6932368721519041699|+|10145330714100251892</td>\n",
       "      <td>776</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP107122.1</td>\n",
       "      <td>6932368721519041699|+|11426489427845291638</td>\n",
       "      <td>777</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP051661.1</td>\n",
       "      <td>6932368721519041699|+|12883103544713419939</td>\n",
       "      <td>776</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP022044.1</td>\n",
       "      <td>6932368721519041699|+|12883103544713419939</td>\n",
       "      <td>767</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP030337.1</td>\n",
       "      <td>6932368721519041699|+|16056383497489254679</td>\n",
       "      <td>760</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP128907.1</td>\n",
       "      <td>6932368721519041699|+|1708810210093552769</td>\n",
       "      <td>760</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP022326.1</td>\n",
       "      <td>6932368721519041699|+|5100696243500885373</td>\n",
       "      <td>777</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP022326.1</td>\n",
       "      <td>6932368721519041699|+|6097212296783908714</td>\n",
       "      <td>777</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP051663.1</td>\n",
       "      <td>6932368721519041699|+|7987278316461198430</td>\n",
       "      <td>764</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP124410.1</td>\n",
       "      <td>6932368721519041699|+|9834685609301298901</td>\n",
       "      <td>768</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP019012.1</td>\n",
       "      <td>6932368721519041699|-|10145330714100251892</td>\n",
       "      <td>768</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_LR890693.1</td>\n",
       "      <td>6932368721519041699|-|11588582069103229002</td>\n",
       "      <td>760</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_OX030701.1</td>\n",
       "      <td>6932368721519041699|-|1314867519434933921</td>\n",
       "      <td>760</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP022044.1</td>\n",
       "      <td>6932368721519041699|-|14506298656834147691</td>\n",
       "      <td>767</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP124455.1</td>\n",
       "      <td>6932368721519041699|-|15558722379768673797</td>\n",
       "      <td>768</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP107122.1</td>\n",
       "      <td>6932368721519041699|-|15558722379768673797</td>\n",
       "      <td>777</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP107162.1</td>\n",
       "      <td>6932368721519041699|-|15558722379768673797</td>\n",
       "      <td>768</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP107151.1</td>\n",
       "      <td>6932368721519041699|-|15558722379768673797</td>\n",
       "      <td>760</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP023826.1</td>\n",
       "      <td>6932368721519041699|-|17115477248436968411</td>\n",
       "      <td>767</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP018970.1</td>\n",
       "      <td>6932368721519041699|-|17115477248436968411</td>\n",
       "      <td>767</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP021454.1</td>\n",
       "      <td>6932368721519041699|-|913783977968963568</td>\n",
       "      <td>777</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP103562.1</td>\n",
       "      <td>7844090243072536214|+|</td>\n",
       "      <td>1446</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP124424.1</td>\n",
       "      <td>964531385716605116|+|</td>\n",
       "      <td>1324</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP128947.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP059130.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP124460.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP076693.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP036245.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP026794.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP027534.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_AP026788.1</td>\n",
       "      <td>964531385716605116|+|]_[16982887268023976258|+|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP010876.1</td>\n",
       "      <td>964531385716605116|-|]_[16982887268023976258|-|</td>\n",
       "      <td>1893</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_LR890693.1</td>\n",
       "      <td>964531385716605116|-|]_[16982887268023976258|-|</td>\n",
       "      <td>1893</td>\n",
       "      <td>segment_1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP124374.1</td>\n",
       "      <td>964531385716605116|-|]_[16982887268023976258|-|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus1</td>\n",
       "      <td>NZ_CP019015.1</td>\n",
       "      <td>964531385716605116|-|]_[16982887268023976258|-|</td>\n",
       "      <td>1863</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus2</td>\n",
       "      <td>NZ_AP022171.1</td>\n",
       "      <td>18228278273347143766|+|</td>\n",
       "      <td>820</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>CIRMBUYJFK_f__CWCCKOQCWZ_r</td>\n",
       "      <td>consensus2</td>\n",
       "      <td>NZ_CP098219.1</td>\n",
       "      <td>2278011600046483881|+|</td>\n",
       "      <td>37622</td>\n",
       "      <td>segment_0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 junction_name   consensus    genome_name  \\\n",
       "27  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP026794.1   \n",
       "28  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP026788.1   \n",
       "9   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP124467.1   \n",
       "29  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP134384.1   \n",
       "36  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP133923.1   \n",
       "41  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP019008.1   \n",
       "12  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP021935.1   \n",
       "13  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP107114.1   \n",
       "14  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP107172.1   \n",
       "18  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP059130.1   \n",
       "19  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP021179.1   \n",
       "10  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP124429.1   \n",
       "16  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP018979.1   \n",
       "15  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP051659.1   \n",
       "39  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP051661.1   \n",
       "17  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP107122.1   \n",
       "30  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP051661.1   \n",
       "34  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP022044.1   \n",
       "32  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP030337.1   \n",
       "26  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP128907.1   \n",
       "24  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP022326.1   \n",
       "42  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP022326.1   \n",
       "45  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP051663.1   \n",
       "33  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP124410.1   \n",
       "35  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP019012.1   \n",
       "5   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_LR890693.1   \n",
       "21  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_OX030701.1   \n",
       "48  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP022044.1   \n",
       "0   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP124455.1   \n",
       "1   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP107122.1   \n",
       "23  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP107162.1   \n",
       "44  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP107151.1   \n",
       "7   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP023826.1   \n",
       "20  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP018970.1   \n",
       "8   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP021454.1   \n",
       "47  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP103562.1   \n",
       "31  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP124424.1   \n",
       "2   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP128947.1   \n",
       "3   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP059130.1   \n",
       "4   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP124460.1   \n",
       "22  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP076693.1   \n",
       "25  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP036245.1   \n",
       "37  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP026794.1   \n",
       "38  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP027534.1   \n",
       "40  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_AP026788.1   \n",
       "6   CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP010876.1   \n",
       "11  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_LR890693.1   \n",
       "43  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP124374.1   \n",
       "46  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus1  NZ_CP019015.1   \n",
       "50  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus2  NZ_AP022171.1   \n",
       "49  CIRMBUYJFK_f__CWCCKOQCWZ_r  consensus2  NZ_CP098219.1   \n",
       "\n",
       "                                     insertion_path  insertion_length  \\\n",
       "27                          16556873922195442803|+|              1661   \n",
       "28                          16556873922195442803|+|              1661   \n",
       "9                             185719230501067033|+|              4204   \n",
       "29                            185719230501067033|+|              4203   \n",
       "36                            185719230501067033|+|              4204   \n",
       "41                            185719230501067033|+|              4204   \n",
       "12                            517157241956952768|+|              1248   \n",
       "13                            517157241956952768|+|              1248   \n",
       "14                            517157241956952768|+|              1248   \n",
       "18                            517157241956952768|+|              1253   \n",
       "19                            517157241956952768|+|              1248   \n",
       "10                            517157241956952768|-|              1248   \n",
       "16                            517157241956952768|-|              1248   \n",
       "15       6932368721519041699|+|10145330714100251892               760   \n",
       "39       6932368721519041699|+|10145330714100251892               776   \n",
       "17       6932368721519041699|+|11426489427845291638               777   \n",
       "30       6932368721519041699|+|12883103544713419939               776   \n",
       "34       6932368721519041699|+|12883103544713419939               767   \n",
       "32       6932368721519041699|+|16056383497489254679               760   \n",
       "26        6932368721519041699|+|1708810210093552769               760   \n",
       "24        6932368721519041699|+|5100696243500885373               777   \n",
       "42        6932368721519041699|+|6097212296783908714               777   \n",
       "45        6932368721519041699|+|7987278316461198430               764   \n",
       "33        6932368721519041699|+|9834685609301298901               768   \n",
       "35       6932368721519041699|-|10145330714100251892               768   \n",
       "5        6932368721519041699|-|11588582069103229002               760   \n",
       "21        6932368721519041699|-|1314867519434933921               760   \n",
       "48       6932368721519041699|-|14506298656834147691               767   \n",
       "0        6932368721519041699|-|15558722379768673797               768   \n",
       "1        6932368721519041699|-|15558722379768673797               777   \n",
       "23       6932368721519041699|-|15558722379768673797               768   \n",
       "44       6932368721519041699|-|15558722379768673797               760   \n",
       "7        6932368721519041699|-|17115477248436968411               767   \n",
       "20       6932368721519041699|-|17115477248436968411               767   \n",
       "8          6932368721519041699|-|913783977968963568               777   \n",
       "47                           7844090243072536214|+|              1446   \n",
       "31                            964531385716605116|+|              1324   \n",
       "2   964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "3   964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "4   964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "22  964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "25  964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "37  964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "38  964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "40  964531385716605116|+|]_[16982887268023976258|+|              1863   \n",
       "6   964531385716605116|-|]_[16982887268023976258|-|              1893   \n",
       "11  964531385716605116|-|]_[16982887268023976258|-|              1893   \n",
       "43  964531385716605116|-|]_[16982887268023976258|-|              1863   \n",
       "46  964531385716605116|-|]_[16982887268023976258|-|              1863   \n",
       "50                          18228278273347143766|+|               820   \n",
       "49                           2278011600046483881|+|             37622   \n",
       "\n",
       "      segment  hits_in_genome  \n",
       "27  segment_1               2  \n",
       "28  segment_1               2  \n",
       "9   segment_0               1  \n",
       "29  segment_0               1  \n",
       "36  segment_0               1  \n",
       "41  segment_0               1  \n",
       "12  segment_0               3  \n",
       "13  segment_0               5  \n",
       "14  segment_0               2  \n",
       "18  segment_0               4  \n",
       "19  segment_0               7  \n",
       "10  segment_0               6  \n",
       "16  segment_0               2  \n",
       "15  segment_0               3  \n",
       "39  segment_1               4  \n",
       "17  segment_1               3  \n",
       "30  segment_0               4  \n",
       "34  segment_0               9  \n",
       "32  segment_0               3  \n",
       "26  segment_0               3  \n",
       "24  segment_0               4  \n",
       "42  segment_1               4  \n",
       "45  segment_0               3  \n",
       "33  segment_0               5  \n",
       "35  segment_0               1  \n",
       "5   segment_0               3  \n",
       "21  segment_0               4  \n",
       "48  segment_1               9  \n",
       "0   segment_0               7  \n",
       "1   segment_0               3  \n",
       "23  segment_0               6  \n",
       "44  segment_0               3  \n",
       "7   segment_0               4  \n",
       "20  segment_0               7  \n",
       "8   segment_0               4  \n",
       "47  segment_0               2  \n",
       "31  segment_0               2  \n",
       "2   segment_0               2  \n",
       "3   segment_1               2  \n",
       "4   segment_0               2  \n",
       "22  segment_0               2  \n",
       "25  segment_0               2  \n",
       "37  segment_0               2  \n",
       "38  segment_0               2  \n",
       "40  segment_0               2  \n",
       "6   segment_0               2  \n",
       "11  segment_1               2  \n",
       "43  segment_0               2  \n",
       "46  segment_0               2  \n",
       "50  segment_0               2  \n",
       "49  segment_0               1  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "insertions_df.sort_values(['consensus','insertion_path'])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "junctions",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
